test/anyof.test.js

```diff
@@ -471,3 +471,26 @@ test('anyOf object with field of type string with format or null', (t) => {
     prop: toStringify
   }), `{"prop":"${toStringify.toISOString()}"}`)
 })
+
+test('anyOf object with field date of type string with format or null', (t) => {
+  t.plan(1)
+  const toStringify = '2011-01-01'
```

Comment:
I would add a test with a `foo bar` string and add a disclaimer in the readme that a date format string, will not be validated (because we assume that the user has already parsed the Date object)